pkg/fieldpath/fieldpath.go

```diff
@@ -122,3 +145,19 @@ func convertResourceMemoryToString(memory *resource.Quantity, divisor resource.Q
 	m := int64(math.Ceil(float64(memory.Value()) / float64(divisor.Value())))
 	return strconv.FormatInt(m, 10), nil
 }
+
+// MergeContainerResourceLimitsWithAllocatable checks if a limit is applied for
+// the container, and if not, it sets the limit based on the node allocatable.
+func MergeContainerResourceLimitsWithAllocatable(container *api.Container,
```

Comment:
nit: MergeContainerResourceLimits --- this method is generic, and should work with any resource list passed in.